internal/copier/copier.go

```
@@ -5,35 +5,330 @@ import (
+			}
+		}
+	}
+	return false
+}
+
+// shouldPreserve checks if a path should be preserved based on exclusion patterns
+// It returns true if:
+// 1. The path is hidden (starts with a dot)
+// 2. The path matches any of the exclusion patterns
+// 3. The path is a directory that contains files matching any of the exclusion patterns
+func shouldPreserve(path string, isDir bool, excludePatterns []string) bool {
```

golangci-lint (1.64.8)

func `shouldPreserve` is unused

(unused)